interop-tests/src/test/scala/akka/grpc/scaladsl/RichErrorModelSpec.scala

```diff
@@ -0,0 +1,123 @@
+import io.grpc.StatusRuntimeException
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.{ Duration, _ }
+import scala.concurrent.{ Await, ExecutionContext, Future }
+
+class RichErrorModelSpec
+    extends TestKit(ActorSystem("RichErrorSpec"))
+    with AnyWordSpecLike
+    with Matchers
+    with BeforeAndAfterAll
+    with ScalaFutures {
+
+  import com.google.protobuf.any.Any
+  import com.google.rpc.{ Code, Status }
+  import io.grpc.protobuf.StatusProto
+
+  implicit val sys: ActorSystem = system
+  implicit val ec: ExecutionContext = sys.dispatcher
+
+  object RichErrorImpl extends GreeterService {
+
+    // #rich_error_model_unary
+    private def toJavaProto(scalaPbSource: com.google.protobuf.any.Any): com.google.protobuf.Any = {
+      val javaPbOut = com.google.protobuf.Any.newBuilder
+      javaPbOut.setTypeUrl(scalaPbSource.typeUrl)
+      javaPbOut.setValue(scalaPbSource.value)
+      javaPbOut.build
+    }
+
+    def sayHello(in: HelloRequest): Future[HelloReply] = {
+      val status: Status = Status
+        .newBuilder()
+        .setCode(Code.INVALID_ARGUMENT.getNumber)
+        .setMessage("What is wrong?")
+        .addDetails(toJavaProto(Any.pack(new HelloReply("The password!"))))
+        .build()
+      Future.failed(StatusProto.toStatusRuntimeException(status))
+    }
+    // #rich_error_model_unary
+
+    override def itKeepsTalking(in: Source[HelloRequest, NotUsed]): Future[HelloReply] = ???
+
+    override def itKeepsReplying(in: HelloRequest): Source[HelloReply, NotUsed] = ???
+
+    override def streamHellos(in: Source[HelloRequest, NotUsed]): Source[HelloReply, NotUsed] = ???
+  }
+
+  val conf = ConfigFactory
+    .load()
+    //.parseString("akka.http.server.preview.enable-http2 = on")
+    .withFallback(ConfigFactory.defaultApplication())
+
+  "Rich error model" should {
+    "work with the manual approach" in {
+
+      // #custom_eHandler
+      val customHandler: PartialFunction[Throwable, Trailers] = {
+        case grpcException: StatusRuntimeException =>
+          Trailers(grpcException.getStatus, new GrpcMetadataImpl(grpcException.getTrailers))
+      }
+
+      val service: HttpRequest => Future[HttpResponse] =
+        GreeterServiceHandler(RichErrorImpl, eHandler = (_: ActorSystem) => customHandler)
+      // #custom_eHandler
+
+      val bound = Await.result(
+        Http(system).newServerAt(interface = "127.0.0.1", port = 0).bind(service),
+        Duration(5, TimeUnit.SECONDS))
+
+      val client = GreeterServiceClient(
+        GrpcClientSettings.connectToServiceAt("127.0.0.1", bound.localAddress.getPort).withTls(false))
+
+      // #client_request
+      val richErrorResponse =
+        Await.result(client.sayHello(HelloRequest("Bob")).failed, Duration(10, TimeUnit.SECONDS))
```

Comment:
It'd be nice to avoid showing `Await.result` in user-facing docs - perhaps we can `foreach`/`map` instead?